analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -176,6 +175,51 @@ public void Execute_PersistSymbols_InsideBlock()
                 ((ILocalReferenceOperation)operation.Instance).Local;
         }
 
+        [TestMethod]
+        public void Execute_UnusedVariable_ClearedAfterBlock()
+        {
+            const string code = @"
+var first = boolParameter ? true : false;
+Tag(""BeforeLastUse""); 
+bool second = first;
+if(boolParameter)
+    boolParameter.ToString();
+Tag(""AfterLastUse"");
+";
+            ISymbol firstSymbol = null;
+            var postProcess = new PostProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.TrackedSymbol() is { } symbol)
+                {
+                    if (symbol.Name == "first")
+                    {
+                        firstSymbol = symbol;
+                    }
+                }
+                return x.State;
+            });
+            var validator = SETestContext.CreateCS(code, postProcess).Validator;
+            validator.TagStates("BeforeLastUse").Should().HaveCount(2).And.OnlyContain(x => x[firstSymbol] != null);
+            validator.TagStates("AfterLastUse").Should().HaveCount(1).And.OnlyContain(x => x[firstSymbol] == null);
+        }
+
+        [TestMethod]
+        public void Execute_CapturedVariable_NotCleared()
+        {
+            const string code = @"
+void LocalFunction()
+{
+boolParameter = false;
+var misc = true;
+  if(misc)
+    misc.ToString();
+Tag(""LocalFunctionEnd"", boolParameter);
+}";
+            var validator = SETestContext.CreateCSForLocalFunction(code, null, "LocalFunction").Validator;
+            validator.TagValues("LocalFunctionEnd").Should().HaveCount(1)
+                .And.ContainSingle(x => x.HasConstraint(BoolConstraint.False));
```

Comment:
```suggestion
            SETestContext.CreateCSForLocalFunction(code, null, "LocalFunction").Validator.TagValues("LocalFunctionEnd")
                .Should().HaveCount(1)
                .And.ContainSingle(x => x.HasConstraint(BoolConstraint.False));
```

Reply:
This will need to be reapplied, as we don't assert the correct thing now - see previous comment.